platform/linux-generic/pktio/loop.c

```diff
@@ -234,6 +236,8 @@ static int loopback_recv(pktio_entry_t *pktio_entry, int index ODP_UNUSED,
 				pkt = new_pkt;
 				pkt_hdr = packet_hdr(new_pkt);
 			}
+
+			_odp_packet_l4_chksum(pkt_hdr, pktio_entry->s.in_chksums, l4_part_sum);
```

Comment:
Why `_odp_packet_parse_common()` cannot perform this? It would make the code quite a lot simpler and easier to maintain.